distro/release_tools.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2019 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Utilities to help create a rule set release."""
+
+import hashlib
+import os
+from string import Template
+import textwrap
+
+
+def package_basename(repo, version):
+  return '%s-%s.tar.gz' % (repo, version)
+
+
+def get_package_sha256(tarball_path):
+  with open(tarball_path, 'rb') as pkg_content:
+    tar_sha256 = hashlib.sha256(pkg_content.read()).hexdigest()
+  return tar_sha256
+
+
+def workspace_content(url, repo, sha256):
+  # Set up a fresh Bazel workspace
+  workspace_stanza_template = Template(textwrap.dedent(
+      """
+      load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+      http_archive(
+          name = "${repo}",
+          url = "${url}",
+          sha256 = "${sha256}",
+      )
+      load("@${repo}//:deps.bzl", "${repo}_dependencies")
```

Comment:
Shouldn't this expand to `load("@rules_java//java:repositories.bzl", ...)`? https://docs.bazel.build/versions/master/skylark/deploying.html#dependencies

Reply:
Ugh. Yes. But that is super confusing. I think the fact that repositories is not at the top level a bug in rules_java. I have to hunt down where we made the the choice to interpose the language name in rules_{language}.  

But, again, the code is in rules_pkg. Next pass through I will add features to to allow overriding path to the dependencies rule.  And the fact that rule sets are all over the map with the words repositories, dependencies and setup is a different issue.